blocks/product-gallery-block/features/product-gallery/_children/ProductFocusView/index.jsx

```diff
@@ -0,0 +1,138 @@
+
+	const shouldShowDownButton = () =>
+		productImages.length >= thumbnailBarItemLimit &&
+		thumbnailBarStartIndex + thumbnailBarItemLimit < productImages.length;
+
+	const shouldShowUpButton = () => thumbnailBarStartIndex !== 0;
+
+	useEffect(() => {
+		let scrollTimeout;
+		const scrollImageIntoView = () => {
+			const currentImageContainer = mainImagesRef.current?.[selectedImageIndex];
+			if (currentImageContainer?.querySelector("img").complete) {
+				currentImageContainer.scrollIntoView();
+			} else {
+				scrollTimeout = setTimeout(scrollImageIntoView, 10);
+			}
+		};
+		scrollImageIntoView();
+		return () => clearTimeout(scrollTimeout);
+	}, [selectedImageIndex, mainImagesRef]);
+
+	return (
+		<div className={`${BLOCK_CLASS_NAME}__focus-view`}>
+			<Stack direction="horizontal" justifacation="start">
+				<div className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-bar`}>
+					<Paragraph>
+						{selectedImageIndex + 1} / {productImages.length}
+					</Paragraph>
+					{shouldShowUpButton() ? (
+						<Button
+							accessibilityLabel="Previous Image"
+							onClick={() => setThumbnailBarStartIndex(thumbnailBarStartIndex - 1)}
+						>
+							<Icon name="ChevronUp" />
+						</Button>
+					) : (
+						<div className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-bar-spacer`} />
+					)}
+					<Stack alignment="center" direction="vertical" justifacation="start">
+						{productImages
+							.filter(
+								(_, index) =>
+									index >=
+										Math.min(
+											thumbnailBarStartIndex,
+											productImages.length - thumbnailBarItemLimit
+										) && index < thumbnailBarStartIndex + thumbnailBarItemLimit
+								// as long as selected index is less than the length of the product images minus the window
+							)
+							.map((image) => (
+								<Image
+									// used as part of a page design so empty string alt text
+									alt=""
+									className={`${BLOCK_CLASS_NAME}__focus-view-thumbnail-image${
+										image._id === productImages[selectedImageIndex]._id ? "--selected" : ""
+									}`}
+									height={100}
+									key={image._id}
+									onClick={() => setSelectedImageIndex(getImageIndexById(image._id))}
+									resizedOptions={{ auth: image.auth[resizerAppVersion] }}
+									resizerURL={resizerURL}
+									responsiveImages={[120]}
+									src={imageANSToImageSrc(image)}
+									width={100}
+								/>
+							))}
+					</Stack>
+					{shouldShowDownButton() ? (
+						<Button
+							accessibilityLabel="Next Image"
+							onClick={() => setThumbnailBarStartIndex(thumbnailBarStartIndex + 1)}
+						>
+							<Icon name="ChevronDown" />
+						</Button>
+					) : null}
+				</div>
+				<Stack
+					alignment="center"
+					direction="vertical"
+					className={`${BLOCK_CLASS_NAME}__focus-view-main-images`}
+				>
+					{productImages.map((image, index) => (
+						<div
+							style={{ flexBasis: "100vh", flexShrink: 0 }}
+							key={image._id}
```

Comment:
`key`s should be unique:
![image](https://user-images.githubusercontent.com/2287238/194131200-723f4b54-7247-498c-9cae-ae42ce1f6c89.png)

Should just be a matter of appending the `_id`: `focus_main_view_${image._id}`

Reply:
@nschubach : Good call! Made this change.